lib/src/signed_video_h26x_auth.c

```diff
@@ -478,7 +546,7 @@ verify_hashes_with_sei(signed_video_t *self, int *num_expected_nalus, int *num_r
   if (self->gop_info->verified_signature_hash == 1) {
     validation_status = '.';
     num_expected_hashes = (int)self->gop_info->num_sent_nalus;
-    if (!verify_gop_hash(self)) {
+    if (!gop_is_ok || !order_ok) {
```

Comment:
I don't think this works. If `gop_is_ok = true` and `order_ok = false`, and you have a hash list, then you will verify with the hash list, which will result in OK, since `gop_is_ok`. Hence, the validation will become OK, when it is not.
One alternative solution is to rewrite this section as
```
validation_status = (gop_is_ok && order_ok) ? '.' : 'N';
num_expected_hashes = (int)self->gop_info->num_sent_nalus;
if (!gop_is_ok && order_ok && self->gop_info->list_idx > 0) {
  // Give validation a second chance if the GOP hash could not be verified, but the order
  // is OK, and the SEI includes a hash list.
  return verify_hashes_with_hash_list(self, num_expected_nalus, num_received_nalus);
 }
```
but note that there are other ways of doing it.